front/lib/swr/swr.ts

```diff
@@ -13,19 +14,68 @@ export function useSWRWithDefaults<TKey extends Key, TData>(
   fetcher: Fetcher<TData, TKey> | null,
   config?: SWRConfiguration & { disabled?: boolean }
 ) {
-  const { mutate: globalMutate } = useSWRConfig();
+  const { mutate: globalMutate, cache } = useSWRConfig();
 
   const mergedConfig = { ...DEFAULT_SWR_CONFIG, ...config };
   const disabled = !!mergedConfig.disabled;
 
   const result = useSWR(disabled ? null : key, fetcher, mergedConfig);
 
+  const mutateKeysWithSameUrl = useCallback(
+    (key: TKey) => {
+      // If the key looks like an url, we need to remove the query params
+      // to make sure we don't cache different pages together
+      // Naive way to detect url by checking for '/'
+      if (typeof key === "string" && key.includes("/")) {
```

Comment:
More robust than this could be something like this:

```
try {
  const url = new URL(myUrl);
} catch (err) {
  // Invalid url.
}
```

Reply:
I can't, it will raise if there is not protocol (tested). I need a way to know before using new URL() on it.